src/userAttributes.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+ * userAttributes.cpp
+ *
+ *  Created on: 26/02/2015
+ *      Author: michael
+ */
+
+#include "userAttributes.h"
+#include <stdexcept>
+#include "sharedbuffer.h"
+#include "megaapi.h"
+
+namespace mega {
+
+void
+UserAttributes::addUserAttribute(std::string &valueName, ValueMap &value,
+        Visibility visibility) {
+    if(tlvStore.find(valueName) != tlvStore.end()) {
+        throw std::runtime_error(ATTRIBUTE_EXISTS);
+    }
+
+    SharedBuffer buffer = this->valueMapToTlv(value, visibility);
+    tlvStore.insert({ valueName, buffer });
+}
+
+ValueMap
+UserAttributes::getUserAttribute(std::string &valueName) {
+
+    auto i = tlvStore.find(valueName);
+    if(i == tlvStore.end()) {
+        throw std::runtime_error(VALUE_NOT_FOUND);
+    }
+
+    ValueMap retVal = this->tlvToValueMap(i->second);
+
+    return retVal;
+}
+
+SharedBuffer
+UserAttributes::getUserAttributeTlv(std::string &valueName) {
+
+    auto i = tlvStore.find(valueName);
+    if(i == tlvStore.end()) {
+        throw std::runtime_error(VALUE_NOT_FOUND);
+    }
+
+    return i->second;
+}
+
+SharedBuffer
+UserAttributes::valueMapToTlv(ValueMap &valueMap, Visibility visibility) {
+    int length = 0;
+        for(auto &i : *valueMap) {
```

Comment:
are there tab indentations in here? Should be 4 spaces indentation levels only, try to avoid tabs (or convert them).